addons/mail/tests/common.py

```diff
@@ -318,7 +318,11 @@ def _find_mail_mail_wemail(self, email_to, status, mail_message=None, author=Non
         """
         filtered = self._filter_mail(status=status, mail_message=mail_message, author=author)
         for mail in filtered:
-            if (mail.email_to == email_to and not mail.recipient_ids) or (not mail.email_to and mail.recipient_ids.email == email_to):
+            # only match one
+            match_email_to = mail.email_to == email_to and not mail.recipient_ids
+            match_recipient = not mail.email_to and mail.recipient_ids.email == email_to
+            match_recipient = match_recipient or not mail.email_to and mail.recipient_ids.email_formatted == email_to
```

Comment:
Isn't it covered by using the 'email_to_recipients' optional argument that allows to differentiate the email used in the mail.mail and the email used in the outgoing email ? That way we assert current state of email used in both records, so that if it changes it is highlighted by the tests.